PinClustering/PinClustering/ViewControllers/MapViewController+MapHandlingExtension.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  MapViewController+MapHandlingExtension.swift
+//  PinClustering
+//
+//  Created by Norbert Agoston on 13/09/16.
+//  Copyright © 2016 Norbert Agoston. All rights reserved.
+//
+
+import Foundation
+import MapKit
+
+extension MapViewController: MKMapViewDelegate {
+    
+    func setupMap(pointsOfInterest: [AnyObject]) {
+        guard let pointOfInterest = pointsOfInterest.first as? FuelLocation else {
+            return
+        }
+        
+        centerMapOnLocation(pointOfInterest.coordinate)
+        displayOnMap(pointsOfInterest)
+    }
+    
+    func displayOnMap(pointsOfInterest: [AnyObject]) {
+        map.addAnnotations(pointsOfInterest as! [FuelLocation])
```

Comment:
try not to force unwrap. Use if let if there is no other way to add the points of interest to the map.